Mage/src/main/java/mage/constants/RoleType.java

```diff
@@ -0,0 +1,50 @@
+package mage.constants;
+
+import mage.abilities.Ability;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.CursedRoleToken;
+import mage.game.permanent.token.Token;
+import mage.game.permanent.token.YoungHeroRoleToken;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * @author TheElk801
+ */
+public enum RoleType {
```

Comment:
I am conflicted on that.
I like the enum grouping with each instance linking to its token, don't get me wrong.
However I do not think this should be categorized/named as a Type.

Reply:
Nevermind, ignore me, this is exactly like TokenType